app/views/userpage.html

```diff
@@ -0,0 +1,279 @@
+    &emsp;&emsp;&emsp;&emsp;<button class="button button2" name="View_Contacts" value="View_Contacts"  onclick="Show_Contacts()" >View Contacts</button>
+</div>
+
+
+<div id="user_page" align="middle">
+    <h1 style="color: darkred; font-size: 28px;
+        font-weight: bold; text-shadow: 1px 1px 1px #555;">Enter new contact!</h1>
+    <form id="contact-form" onsubmit="AddContact(); return false">
+            <p><label>First Name    : &emsp;&emsp;<input type="text" name="first-name" required="enabled"></label></p>
+            <p><label>Last Name     : &emsp;&emsp;<input type="text" name="last-name" required="enabled"></label></p>
+            <p><label>E-mail Address : <input type=email name="email" required="enabled"></label></p>
+            <p>&emsp;&emsp;&emsp;&emsp;&emsp;<label>Phone Number  : <input name="phone1" required="enabled"></label>
+            <button class="button button2"  name="add-number" value="add_number"  onclick="add_number(); return false;" > + </button></p><br>
+            <input class="button button2" type="submit" value="Add-Contact"  >
+    </form>
+</div>
+<br>
+<div id="contact-table" style="display: none;">
+    <table>
+        <thead>
+        <tr style="text-align: left;">
+            <th width="20%" style="color: darkblue;">First Name</th>
+            <th width="20%" style="color: darkblue;">Last Name</th>
+            <th width="30%" style="color: darkblue;">E-mail</th>
+            <th width="20%" style="color: darkblue;">Phone Number</th>
+            <th width="10%" ></th>
+        </tr>
+        </thead>
+        <tbody id="view-contacts">
+        {{range .MyUser.Contacts}}
+        <tr id="contact-row-{{.Id}}">
+            <td>{{.FirstName}}</td>
+            <td>{{.LastName}}</td>
+            <td>{{.Email}}</td>
+            <td>{{range .PhoneNumber}} <object id="number-{{.ID}}"> {{.Phonenumber}} <button class="delete-btn" onclick="DeleteNum({{.ID}})" >Delete</button> </object><br>{{end}}</td>
+
+            <td> <button class="delete-btn" onclick="Delete({{.Id}})" >Delete</button></td>
+        </tr>
+        {{end}}
+
+
+        </tbody>
+
+
+    </table>
+</div>
+</div>
+
+
+
+
+<script>
+
+  function add_number()
+    {
+        var html_text = document.createTextNode("Number: ");
+        var html_element =document.createElement( "input");
+        var att = document.createAttribute("name");
+        count= count+1;
+        att.value='phone' + count;
+        html_element.setAttributeNode(att);
+        var theDiv =  document.getElementById("numbers");
+        theDiv.appendChild(html_text);
+        theDiv.appendChild(html_element);
+
+
+    }
+
+
+function AddContact()
+    {
+
+       $.ajax({
+            url: '/addcontact',
+            type: 'POST',
+            data: $("#contact-form").serialize(),
+            success: function(data){
+
+            location.reload(true);
```

Comment:
This is not the best practice from performance wise that you refresh all html page and get all the content of it again just add the added contact to the table using append